controllers/keda/hpa.go

```diff
@@ -90,10 +91,26 @@ func (r *ScaledObjectReconciler) newHPAForScaledObject(ctx context.Context, logg
 		labels[key] = value
 	}
 
+	minReplicas := getHPAMinReplicas(scaledObject)
+	maxReplicas := getHPAMaxReplicas(scaledObject)
+
+	pausedCount, err := executor.GetPausedReplicaCount(scaledObject)
+	if err != nil {
+		return nil, err
+	}
+	if pausedCount != nil {
+		// MinReplicas on HPA can't be 0
```

Comment:
MinReplicas on HPA can be 0

Reply:
It can be but it's behind a feature gate. From https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/:

> HPAScaleToZero: Enables setting minReplicas to 0 for HorizontalPodAutoscaler resources when using custom or external metrics.

Reply:
In my testing, the changes are rejected if you try to set minReplicas to 0.

Reply:
You are right, sorry for the confusion. It is the other way round, scaling is paused if the target workload is scaled to 0.